network/src/peers/peer_info.rs

```diff
@@ -39,6 +40,8 @@ pub enum PeerStatus {
 
 #[derive(Debug, Default)]
 pub struct PeerQuality {
+    /// The current block height of this peer.
+    pub block_height: RwLock<BlockHeight>,
```

Comment:
since `BlockHeight` is a `u32`, this should just be an `AtomicU32` for better performance

Reply:
Resolved in post networking